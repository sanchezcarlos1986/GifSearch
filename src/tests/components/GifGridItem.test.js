import React from 'react';
import GifGridItem from '../../components/GifGridItem';
import {shallow} from 'enzyme';

describe('GifGridItem', () => {
  let props;
  let component;

  beforeEach(() => {
    props = {
      title: 'Mega Man',
      url: 'http://myurl.com',
    };
    component = shallow(<GifGridItem {...props} />);
  });

  it('should renders ok', () => {
    expect(component).not.toBeNull();
    expect(component.hasClass('GifGridItem')).toBe(true);
    expect(component).toMatchSnapshot();
  });

  it('should renders the image', () => {
    expect(component.find('img').exists()).toBe(true);
    expect(component.find('img').props().src).toBe(props.url);
    expect(component.find('img').props().alt).toBe(props.title);
  });

  it('should renders the title', () => {
    expect(component.find('p').exists()).toBe(true);
    expect(component.find('p').text()).toBe(props.title);
  });
});
