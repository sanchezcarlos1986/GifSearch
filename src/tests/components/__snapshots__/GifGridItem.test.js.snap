// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem should renders ok 1`] = `
<div
  className="GifGridItem"
>
  <img
    alt="Mega Man"
    src="http://myurl.com"
  />
  <p>
    Mega Man
  </p>
</div>
`;
